### ISSUE TITLE
MySQL Error 1054: missing `config.id` and `config.deleted` columns

### ISSUE THREAD

**Issue by pgorod** [salesagility/SuiteCRM-Core#154]
#### Issue
I've grown accustomed to seeing this kind of errors a lot in the Forums, it's definitely a bug that should be looked at...

#### Expected Behavior
No MySql errors.

#### Actual Behavior
Have a look at this list of issues:

https://community.suitecrm.com/search?q=config.deleted

#### Possible Fix
Just guessing: although `config` is a special table, with a special structure, some database class somewhere is treating it like a regular SuiteCRM table, and adding clauses like `AND config.deleted = 0` which won't work.

#### Your Environment
* SuiteCRM Version used: SuiteCRM 8, several versions.

Some of those Forums issues are very old, like 2014! This error seems to have resurfaced after being fixed ([here](https://github.com/salesagility/SuiteCRM/commit/27bd7606f539e7bed55873b4343770c10a5ec05a))

**Comment by sebastianbrosch** [comment]
I can confirm this issue (for SuiteCRM 8.2.0 and SuiteCRM 8.1.3). I found some failed queries in the log (Admin -> System Settings -> View Log). Especially this query failed a lot:

```
SELECT favorites.*
FROM favorites
JOIN config ON ( config.id = favorites.parent_id )
WHERE favorites.assigned_user_id = '1'
  AND favorites.parent_type = 'Administration'
  AND favorites.deleted = 0
  AND config.deleted = 0
ORDER BY favorites.date_entered DESC LIMIT 0,10
```

with this error message:

```
MySQL error 1054: Unknown column 'config.deleted' in 'where clause'
```

**Comment by CommuniG8** [comment]
Hi, I've just installed 8.2.3 for a client and I'm seeing the error.
Is anything being done to fix it?
Is there a work around, e.g. creating the id and deleted columns?
Thanks.

**Comment by philippelt** [comment]
The problem is located in the module /public/legacy/include/portability/Services/Favorites/FavoritesManagerPort.php at line 142.

The favorite manager has been refactored to become generic for any object but the "config" object is missing "id" and "deleted" properties that are now required by this favorites manager.

It should not be too hard to fix but I am far from being a php developer so I am not able to propose a fix.

**Comment by ggraynoth** [comment]
Exactly the same issue here. I've found the source of the problem which @philippelt identified but I have no idea on a fix in regards to coding. As this is impacting aspects such as Campaigns, which are a key function for many users, this is something that does need fixing as a matter of urgency.

**For the benefit of the developers, to understand the business impact.** We have had to suspend campaigns for now and are running them outside of SuiteCRM, via MS Word and mailmerge with Outlook, which is a major backward step and very time consuming, when the data, templates and target lists are all in SuiteCRM. We have effectively had to recreate them externally and export lists and so on. A real pain and with no bounceback management, contact history entries or marketing tracking, all of which is being done manually for now..
